pkg/utils/annotations/annotations_test.go

```diff
@@ -305,3 +304,82 @@ func TestExtractNumberOfResourceSnapshots(t *testing.T) {
 		})
 	}
 }
+
+func TestExtractNumberOfEnvelopeObjFromResourceSnapshot(t *testing.T) {
+	testCases := []struct {
+		name      string
+		snapshot  *fleetv1beta1.ClusterResourceSnapshot
+		want      int
+		wantError bool
+	}{
+		{
+			name: "valid annotation",
+			snapshot: &fleetv1beta1.ClusterResourceSnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: snapshotName,
+					Annotations: map[string]string{
+						fleetv1beta1.NumberOfEnvelopedObjectsAnnotation: "0",
+					},
+				},
+			},
+			want: 0,
+		},
+		{
+			name: "valid annotation",
+			snapshot: &fleetv1beta1.ClusterResourceSnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: snapshotName,
+					Annotations: map[string]string{
+						fleetv1beta1.NumberOfEnvelopedObjectsAnnotation: "1",
+					},
+				},
+			},
+			want: 1,
+		},
+		{
+			name: "no annotation means no enveloped objects",
+			snapshot: &fleetv1beta1.ClusterResourceSnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: snapshotName,
+				},
+			},
+			want: 0,
+		},
+		{
+			name: "invalid annotation: not an integer",
+			snapshot: &fleetv1beta1.ClusterResourceSnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: snapshotName,
+					Annotations: map[string]string{
+						fleetv1beta1.NumberOfEnvelopedObjectsAnnotation: "abc",
+					},
+				},
+			},
+			wantError: true,
+		},
+		{
+			name: "invalid annotation: negative integer",
+			snapshot: &fleetv1beta1.ClusterResourceSnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: snapshotName,
+					Annotations: map[string]string{
+						fleetv1beta1.NumberOfEnvelopedObjectsAnnotation: "-1",
+					},
+				},
+			},
+			wantError: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			got, err := ExtractNumberOfEnvelopeObjFromResourceSnapshot(tc.snapshot)
+			if gotErr := err != nil; gotErr != tc.wantError {
+				t.Fatalf("ExtractNumberOfEnvelopeObjFromResourceSnapshot teset `%s` got err %v, want err %v", tc.name, err, tc.wantError)
```

Comment:
```suggestion
				t.Fatalf("ExtractNumberOfEnvelopeObjFromResourceSnapshot() got err %v, want err %v", tc.name, err, tc.wantError)
```

The tc.name will be always printed as we define it in the t.Run(t.name, xxx)